sdk/cosmosdb/cosmos/src/request/CosmosDiagnostics.ts

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { v4 } from "uuid";
+import { Constants } from "../common";
+import { CosmosHeaders } from "../queryExecutionContext";
+
+/**
+ * Cosmos Diagnostic type. It contains diagnostic information during a client operation. ie. Item.read().
+ * This `clientSideRequestStatistics` member all the diagnostic information related to client process. i.e
+ * - metadata lookups.
+ * - retries
+ * - endpoints contacted.
+ * - request, response payload stats.
+ * Here all the server requests, apart from the final intended resource are considered as metadata calls.
+ * i.e. for item.read(id), if the client makes server call to discover endpoints it would be considered
+ * as metadata call.
+ */
+export interface CosmosDiagnostics {
```

Comment:
Should we name this as CosmosRequestDiangostics?

Reply:
Sajee, from what I understand, `CosmosDiagnostics` is an uber object for all kinds of diagnostic, for example `clientSideRequestStatistics: ClientSideRequestStatistics` is part of `CosmosDiagnostics`. Which is specifically made of client-side request related stats. In other SDK system related information is also part of `CosmosDiagnostics` if in future we want to do similar thing, `CosmosDiagnostics` name would be more appropriate.

Moreover, currently queryStats is part of FeedResponse directly, I have not changed that behavior (ie. not moved queryStats to CosmosDiagnostics), Is it okay? because if we move it to CosmosDiagnostics, it is definitely not client side, but server side stats.